src/transformers/models/unispeech/modeling_unispeech.py

```diff
@@ -1525,3 +1560,319 @@ def forward(
+            param.requires_grad = False
+
+    def pit_loss_single_permutation(self, logits, loss_mask, labels):
+        bce_loss = BCEWithLogitsLoss(reduction="none")
+        loss = bce_loss(logits, labels)
+        loss = loss * loss_mask
+        loss = loss.mean(dim=2).sum(dim=1)
+        loss = torch.unsqueeze(loss, dim=1)
+        return loss
+
+    def pit_loss(self, logits, attention_mask, labels):
+        label_permutations = permutations(range(self.config.num_labels))
+        loss_mask = self._get_feature_vector_attention_mask(logits.shape[1], attention_mask)
+        losses = []
+        for p in label_permutations:
+            labels_perm = labels[:, :, p]
+            loss_perm = self.pit_loss_single_permutation(logits, loss_mask, labels_perm)
+            losses.append(loss_perm)
+        loss = torch.cat(losses, dim=1)
+        min_loss, min_idx = loss.min(dim=1)
+        loss = min_loss.sum() / loss_mask.sum()
+        return loss
+
+    @add_start_docstrings_to_model_forward(UNISPEECH_INPUTS_DOCSTRING)
+    @add_code_sample_docstrings(
+        processor_class=_SEQ_CLASS_PROCESSOR_FOR_DOC,
+        checkpoint=_SEQ_CLASS_CHECKPOINT,
+        output_type=TokenClassifierOutput,
+        config_class=_CONFIG_FOR_DOC,
+        modality="audio",
+    )
+    def forward(
+        self,
+        input_values,
+        attention_mask=None,
+        output_attentions=None,
+        output_hidden_states=None,
+        return_dict=None,
+        labels=None,
+    ):
+        r"""
+        labels (:obj:`torch.LongTensor` of shape :obj:`(batch_size,)`, `optional`):
+            Labels for computing the sequence classification/regression loss. Indices should be in :obj:`[0, ...,
+            config.num_labels - 1]`. If :obj:`config.num_labels == 1` a regression loss is computed (Mean-Square loss),
+            If :obj:`config.num_labels > 1` a classification loss is computed (Cross-Entropy).
+        """
+
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        output_hidden_states = True if self.config.use_weighted_layer_sum else output_hidden_states
+
+        outputs = self.unispeech(
+            input_values,
+            attention_mask=attention_mask,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        if self.config.use_weighted_layer_sum:
+            hidden_states = outputs[_HIDDEN_STATES_START_POSITION]
+            hidden_states = torch.stack(hidden_states, dim=1)
+            norm_weights = nn.functional.softmax(self.layer_weights, dim=-1)
+            hidden_states = (hidden_states * norm_weights.view(-1, 1, 1)).sum(dim=1)
+        else:
+            hidden_states = outputs[0]
+
+        logits = self.classifier(hidden_states)
+
+        loss = None
+        if labels is not None:
+            loss = self.pit_loss(logits, attention_mask, labels)
```

Comment:
Think we don't even need the loss function in a first step as we'll not provide an example script here

Reply:
But if it works already correctly - happy to leave it